lib/sqlalchemy/dialects/mssql/base.py

```diff
@@ -2731,6 +2732,7 @@ def get_indexes(self, connection, tablename, dbname, owner, schema, **kw):
                 "name": row["name"],
                 "unique": row["is_unique"] == 1,
                 "column_names": [],
+                "dialect_options": {"mssql_where": row["filter_definition"]},
```

Comment:
shouldn't `dialect_options` be added only when needed like in postgres?

Reply:
Thanks @CaselIT I have now amended the code to only include dialect_options when needed